src/coreclr/pal/src/arch/riscv64/context2.S

```diff
@@ -9,4 +9,238 @@
+    fld  f24, (CONTEXT_F24)(t0)
+    fld  f25, (CONTEXT_F25)(t0)
+    fld  f26, (CONTEXT_F26)(t0)
+    fld  f27, (CONTEXT_F27)(t0)
+    fld  f28, (CONTEXT_F28)(t0)
+    fld  f29, (CONTEXT_F29)(t0)
+    fld  f30, (CONTEXT_F30)(t0)
+    fld  f31, (CONTEXT_F31)(t0)
+
+    lw t1, (CONTEXT_FLOAT_CONTROL_OFFSET)(t0)
+    fscsr x0, t1
+
+LOCAL_LABEL(No_Restore_CONTEXT_FLOATING_POINT):
+
+    lw  t1, CONTEXT_ContextFlags(t4)
+    andi t1, t1, 0x1 << CONTEXT_INTEGER_BIT
+    beqz t1, LOCAL_LABEL(No_Restore_CONTEXT_INTEGER)
+
+    ld  tp, (CONTEXT_Tp)(a0)
+    ld  gp, (CONTEXT_Gp)(a0)
+    ld  a1, (CONTEXT_A1)(a0)
+    ld  a2, (CONTEXT_A2)(a0)
+    ld  a3, (CONTEXT_A3)(a0)
+    ld  a4, (CONTEXT_A4)(a0)
+    ld  a5, (CONTEXT_A5)(a0)
+    ld  a6, (CONTEXT_A6)(a0)
+    ld  a7, (CONTEXT_A7)(a0)
+    ld  t0, (CONTEXT_T0)(a0)
+    ld  t1, (CONTEXT_T1)(a0)
+    ld  t2, (CONTEXT_T2)(a0)
+    ld  t3, (CONTEXT_T3)(a0)
+    ld  t5, (CONTEXT_T5)(a0)
+    ld  t6, (CONTEXT_T6)(a0)
+
+    ld  s1, (CONTEXT_S1)(a0)
+    ld  s2, (CONTEXT_S2)(a0)
+    ld  s3, (CONTEXT_S3)(a0)
+    ld  s4, (CONTEXT_S4)(a0)
+    ld  s5, (CONTEXT_S5)(a0)
+    ld  s6, (CONTEXT_S6)(a0)
+    ld  s7, (CONTEXT_S7)(a0)
+    ld  s8, (CONTEXT_S8)(a0)
+    ld  s9, (CONTEXT_S9)(a0)
+    ld  s10, (CONTEXT_S10)(a0)
+    ld  s11, (CONTEXT_S11)(a0)
+
+    ld  a0, (CONTEXT_A0)(a0)
+
+LOCAL_LABEL(No_Restore_CONTEXT_INTEGER):
+
+    lw  t1, CONTEXT_ContextFlags(t4)
+    andi t1, t1, 0x1 << CONTEXT_CONTROL_BIT
+    beqz t1, LOCAL_LABEL(No_Restore_CONTEXT_CONTROL)
+
+    ld  ra, (CONTEXT_Ra)(t4)
+    ld  fp, (CONTEXT_Fp)(t4)
+    ld  sp, (CONTEXT_Sp)(t4)
+    ld  t1, (CONTEXT_Pc)(t4) // Since we cannot control $pc directly, we're going to corrupt t1
+    ld  t4, (CONTEXT_T4)(t4)
+    jr  t1
+
+LOCAL_LABEL(No_Restore_CONTEXT_CONTROL):
+    ld  t4, (CONTEXT_T4)(t4)
+    ret
+LEAF_END RtlRestoreContext, _TEXT
+
+// Incoming:
+//  a0: Context*
+
+LEAF_ENTRY RtlCaptureContext, _TEXT
+    PROLOG_STACK_ALLOC 16
+    .cfi_adjust_cfa_offset 16
+    sd  t1, 0(sp)
+    li  t1, CONTEXT_FULL
+    sw  t1, CONTEXT_ContextFlags(a0)
+    ld  t1, 0(sp)
+    EPILOG_STACK_FREE 16
+    .cfi_adjust_cfa_offset -16
+    tail CONTEXT_CaptureContext
+LEAF_END RtlCaptureContext, _TEXT
+
+// Incoming:
+//  a0: Context*
+//
+
+LEAF_ENTRY CONTEXT_CaptureContext, _TEXT
+    PROLOG_STACK_ALLOC 24
+    .cfi_adjust_cfa_offset 24
+    sd  t0, 0(sp)
+    sd  t1, 8(sp)
+    sd  t3, 16(sp)
+
+    lw  t1, CONTEXT_ContextFlags(a0)
+    li  t0, CONTEXT_CONTROL
+    and  t3, t1, t0
+    bne t3, t0, LOCAL_LABEL(Done_CONTEXT_CONTROL)
+
+    addi  t0, sp, 24
+    sd  fp, CONTEXT_Fp(a0)
+    sd  t0, CONTEXT_Sp(a0)
+    sd  ra, CONTEXT_Ra(a0)
+    sd  ra, CONTEXT_Pc(a0)
+
+LOCAL_LABEL(Done_CONTEXT_CONTROL):
+
+    li  t0, CONTEXT_INTEGER
+    and  t3, t1, t0
+    bne  t3, t0, LOCAL_LABEL(Done_CONTEXT_INTEGER)
+
+    ld  t0, 0(sp)
+    ld  t1, 8(sp)
+    ld  t3, 16(sp)
+
+    sd  tp, (CONTEXT_Tp)(a0)
+    sd  gp, (CONTEXT_Gp)(a0)
+    sd  a0, (CONTEXT_A1)(a0)
```

Comment:
Should not this be `CONTEXT_A0` instead of `CONTEXT_A1`?

Reply:
Thank you. It is my mistake.